### ISSUE TITLE
import_to_elastic - Content-Type header is Missing 

### ISSUE THREAD

**Issue by hsaif** [inventaire/wikidata-subset-search-engine#4]
As shown in the documentation, I'm running the command below to filter and index Wikidata.json

`cat wikidata-dump.json.gz | gzip -d | wdfilter --claim P31:Q5 --omit type,sitelinks | import_to_elastic humans`

However, I'm getting this error:

```
****** bulk post err body ******
{"error":"Content-Type header [] is not supported","status":406}
-----
12100
12200
12300
12400
12500
12600
12700
12800
12900
****** putting batch... ******
1000
-----
13000
****** bulk post err ******
Error: Not Acceptable
    at handleHttpErrorCode (/Users/hs/BigData/wikidata-subset-search-engine/node_modules/bluereq/lib/promisified_request.js:27:15)
    at Request._callback (/Users/hs/BigData/wikidata-subset-search-engine/node_modules/bluereq/lib/promisified_request.js:20:9)
    at Request.self.callback (/Users/hs/BigData/wikidata-subset-search-engine/node_modules/request/request.js:186:22)
    at Request.emit (events.js:127:13)
    at Request.<anonymous> (/Users/hs/BigData/wikidata-subset-search-engine/node_modules/request/request.js:1163:10)
    at Request.emit (events.js:127:13)
    at Gunzip.<anonymous> (/Users/hs/BigData/wikidata-subset-search-engine/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:219:13)
    at Gunzip.emit (events.js:132:15)
    at endReadableNT (_stream_readable.js:1101:12)
    at process._tickCallback (internal/process/next_tick.js:152:19)

-----
```

I assume the header should be JSON, but I'm not sure why it's empty

Please advise

**Comment by maxlath** [comment]
Are you getting this error for every batch or did that happen only for that `12000-13000` batch?

It looks like [something that happens with ElasticSearch `>= v6`](https://github.com/taskrabbit/elasticsearch-dump/issues/350), while I have been developing this repo targeting `v2`.

A possible fix could be to replace [this line](https://github.com/inventaire/entities-search-engine/blob/dfea2ca/lib/bulk_post_to_elasticsearch.coffee#L20) by
```coffee
breq.post
  url: "#{elasticHost}/_bulk"
  headers: { 'Content-Type': 'application/json' }
  body: bulk.joinLines batch
```

**Comment by hsaif** [comment]
Thanks Max!
I did replace the code and now it's working fine. 
The process completed, but nothing were imported to ES!

Any ideas please? 

```
****** putting batch... ******
1000
-----
101000
done!
```

**Comment by maxlath** [comment]
that's weird that you reach only 101000, last time I imported all humans from wikidata (`--claim P31:Q5 `), I got over 3 million items, and it took several hours to load oO

**Comment by hsaif** [comment]
I ran it only on a subset of the wikidata (101000 objects). But I also ran it again on 3M objects 
But nothing was indexed by ES :(

so basically I replaced:
```
breq.post "#{elasticHost}/_bulk", bulk.joinLines(batch)
  .then bulk.logRes('bulk post res')
  .catch _.Error('bulk post err')
```

with:

```
breq.post
  url: "#{elasticHost}/_bulk"
  headers: { 'Content-Type': 'application/json' }
  body: bulk.joinLines batch
```

Is there anything else I need to do/check?

**Comment by maxlath** [comment]
I meant to replace by:
```coffee
breq.post
  url: "#{elasticHost}/_bulk"
  headers: { 'Content-Type': 'application/json' }
  body: bulk.joinLines batch
.then bulk.logRes('bulk post res')
.catch _.Error('bulk post err')
```
to log responses and errors, but I guess an error here would have crash the whole promise chain without this catch, so I hardly see how that could be it

**Comment by hsaif** [comment]
Thanks Max, 

You are right. 
Here is the error I'm getting now:

```
{ error:
   { root_cause: [ [Object] ],
     type: 'illegal_argument_exception',
     reason: 'The bulk request must be terminated by a newline [\n]' },
  status: 400 }
```

I also tried `application/x-ndjson`  as suggested by [ES](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/docs-bulk.html) But didn't work either!
I assume a new line [\n] should be added somewhere in the code 

Edit: Just noticed that `joinLines` does add a new line at the end. So I'm not what causes the error!?

**Comment by maxlath** [comment]
could you try to edit `/Users/hs/BigData/wikidata-subset-search-engine/node_modules/request/index.js` by adding `console.log(params)` before `return new request.Request(params)` (line 54, if we have the exact same version)?
that could allow to see what exactly is sent to ElasticSearch

**Comment by hsaif** [comment]
Output Below:
I removed most of the `body` from the output, for clarity

```
****** putting batch... ******
1000
-----
1000
{ url: 'http://localhost:9200/_bulk',
  headers: { 'Content-Type': 'application/json' },
  body: '{"index":{"_index":"wikidata","_type":"entity_slim","_id":"Q23"}}\n{"id":"Q23","labels":{"en":"George Washington"},"descriptions":{"en":"firs[....]}\n',
  json: true,
  gzip: true,
  method: 'POST',
  callback: [Function] }
****** bulk post err ******
Error: Bad Request
    at handleHttpErrorCode (/Users/hs/hs/wikidata-subset-search-engine/node_modules/bluereq/lib/promisified_request.js:27:15)
    at Request._callback (/Users/hs/BigData/wikidata-subset-search-engine/node_modules/bluereq/lib/promisified_request.js:20:9)
    at Request.self.callback (/Users/hs/BigData/wikidata-subset-search-engine/node_modules/request/request.js:186:22)
    at Request.emit (events.js:127:13)
    at Request.<anonymous> (/Users/hs/BigData/wikidata-subset-search-engine/node_modules/request/request.js:1163:10)
    at Request.emit (events.js:127:13)
    at Gunzip.<anonymous> (/Users/hs/BigData/wikidata-subset-search-engine/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:219:13)
    at Gunzip.emit (events.js:132:15)
    at endReadableNT (_stream_readable.js:1101:12)
    at process._tickCallback (internal/process/next_tick.js:152:19)

-----
****** bulk post err body ******
{ error: 
   { root_cause: [ [Object] ],
     type: 'illegal_argument_exception',
     reason: 'The bulk request must be terminated by a newline [\n]' },
  status: 400 }
-----
```

**Comment by maxlath** [comment]
you could try to do the same request from curl, and tweak it until it works?
the request you are doing above corresponds to
```sh
curl http://localhost:9200/_bulk -H 'Content-Type: application/json' -d '{"index":{"_index":"wikidata","_type":"entity_slim","_id":"Q23"}}
{"id":"Q23","labels":{"en":"George Washington"},"descriptions":{}}
'
```

**Comment by hsaif** [comment]
The above request worked fine:

`{"took":17,"errors":false,"items":[{"index":{"_index":"wikidata","_type":"entity_slim","_id":"Q23","_version":2,"result":"updated","_shards":{"total":2,"successful":1,"failed":0},"_seq_no":1,"_primary_term":1,"status":200}}]}`

**Comment by Reasno** [comment]
I saw the same error on Elasticsearch 6.2 (illegal_argument_exception). The code runs fine in my 5.4 setup though.

**Comment by Reasno** [comment]
@hsaif After some investigation, I realized the internal http request lib swallowed the newline in the end. This behavior can be found in [this line](https://github.com/maxlath/bluereq/blob/686d47379977ec8662d20b88310eb26f2281ad5c/lib/get_options.js#L14). 

The following fix is proven to work:
```
breq.post
  url: "#{elasticHost}/_bulk"
  headers: { 'Content-Type': 'application/json' }
  body: bulk.joinLines batch
  json: false
.then bulk.logRes('bulk post res')
.catch _.Error('bulk post err')
```

**Comment by maxlath** [comment]
@Reasno I integrated your fix in eeb5ab5
